src/Microsoft.AspNetCore.Blazor.Server/AutoRebuild/ProcessUtils.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Blazor.Server.AutoRebuild
+{
+    internal static class ProcessUtils
+    {
+        // Based on https://stackoverflow.com/a/3346055
+
+        public static Process GetParent(Process process)
```

Comment:
Interestingly enough in .NET on windows the process APIs don't create a parent-child releationship between processes when you use `System.Process` to launch a new process. Not an issue here since VS does the right thing, but in case we want to use this technique elsewhere, that will come up.

Reply:
Good to know, thanks.

This is only a way of making sure that if you have multiple VS instances open and showing your project, we only ask the correct one to do the build. We could instead have VS write something like a random GUID into the build output to indicate its identity. But since the PID technique appears to work fine it's probably simpler.

Reply:
Yeah I think the approach here is the right one. I just wanted to point out that this only works because VS uses P/Inovkes to launch the process. Anyone writing tools on .NET is familiar with this old problem 😢